File: Trapezoidal.c
#include<stdio.h>

double function( double x){
	// function which has be integrated

	return x*x;	
	}

int main(){
	
	double lowerlimit,upperlimit,stepsize,integral,upper,lower,partitions;
		
	printf("Partitions : ");
	scanf("%lf",&partitions);	
	printf("Lower limit : ");
	scanf("%lf",&lowerlimit);
	printf("Upper limit : ");
	scanf("%lf",&upperlimit);
	
	stepsize = (upperlimit-lowerlimit)/partitions;
	
	upper=lowerlimit;
	lower=lowerlimit;
	integral=0;
	
	for(int i=0;i<partitions;i++){
		upper+=stepsize;
		integral += stepsize*(function(lower)+function(upper))/2;
		lower+=stepsize;	
	}
	printf("Value : %.10f\n",integral);

}

File: Trapezoidal.py
# Function which has to be integrated
function = lambda x : x**2

#input the number of partitions of domain;must be > 1
fractions = int(input("Partitions : "))
#Input lower limit
lowerlimit = int(input("Lower limit : "))
#Input upper limit
upperlimit = int(input("Upper limit : "))
stepsize = (upperlimit-lowerlimit)/fractions




integral=0
upper=lowerlimit
lower=lowerlimit
for i in range(fractions):
	upper+= stepsize
	integral+=(function(upper)+ function(lower))*stepsize/2
	lower+=stepsize
print("Value : ",integral)



